k9mail/src/main/java/com/fsck/k9/fragment/KeywordAddDialogFragment.java

```diff
@@ -0,0 +1,133 @@
+package com.fsck.k9.fragment;
+
+import java.lang.CharSequence;
+import java.lang.ClassCastException;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.app.DialogFragment;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import com.fsck.k9.mail.Keyword;
+import com.fsck.k9.R;
+
+public class KeywordAddDialogFragment extends DialogFragment {
+
+    public interface KeywordAddDialogFragmentDialogListener {
+        public void onKeywordAdded(String externalCode);
+    }
+
+    private EditText externalCodeEdit;
+    private TextView message;
+    private KeywordAddDialogFragmentDialogListener listener;
+    private Button positiveButton;
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        outState.putSerializable(
+            "external_code", externalCodeEdit.getText().toString());
+    }
+
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstance) {
+
```

Comment:
As a first step I have extracted the anonymous classes. Please let me know if you would like me to break it down further.